src/types.rs

```diff
@@ -117,6 +119,27 @@ impl RowEntry {
     }
 }
 
+impl Eq for RowEntry {}
+
+impl PartialEq<Self> for RowEntry {
+    fn eq(&self, other: &Self) -> bool {
+        self.key == other.key && self.seq == other.seq
+    }
+}
+
+impl PartialOrd<Self> for RowEntry {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for RowEntry {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // Compare by key first, then by sequence number
+        (&self.key, self.seq).cmp(&(&other.key, other.seq))
```

Comment:
i'm concerned that this approach should also be used with caution. in most cases, we need to prioritize the row entry with the highest sequence number in front. i'm kinda prefer to implement the ordering where needed locally with test covered with local comments.